### ISSUE TITLE
ASP.NET Core 2.1 high-level planning for docs - master

### ISSUE THREAD

**Issue by Rick-Anderson** [aspnet/Docs#5404]
**Last Updated: 10/29/2018**

Copy from aspnet/Announcements#288 - order rearranged by priority.
[ASP.NET Core 2.1 roadmap blog](https://blogs.msdn.microsoft.com/webdev/2018/02/02/asp-net-core-2-1-roadmap/)
[2.1 Project Page](https://github.com/aspnet/Docs/projects/9)

The following high-level features are planned for the ASP.NET Core 2.1 release:

- [x] .NET Core meta doc, add ASP.NET Core features
- [x] SignalR #5389 @rachelappel - Real-time web framework on ASP.NET Core (aspnet/SignalR#394)
- [x] HTTPS #5393, @rick-anderson - On by default and easy to configure (aspnet/Home#2308)
- [x] Add HttpsRedirection and HSTS middleware documentation #4568 @Rick-Anderson
- [ ] HTTPS  #3310 @scottaddie - How to create cert xplat, config Kestresl to use that cert in prod/dev. etc
- [x] IHttpClientFactory #5402- @scottaddie HttpClient as a service, handle cross-cutting concerns like caching, retry 
- [x] UI as a library #5409 - @Rick-Anderson -  Package Razor pages and views as reusable libraries (aspnet/Razor#1809) Check stability before starting.
- [x] Identity UI package and scaffolder #5410 -@Rick-Anderson - Add identity to any application (aspnet/Home#2311)
- [x] GDPR compliance cookies #5400 - Templates updated with new privacy features (aspnet/Security#1561, aspnet/Identity#1341) @rick-anderson
- [x] GDPR delete custom Identity user data. #6226 @rick-anderson
- [x] Web API conventions #5401 - @scottaddie - Rich Swagger support without attributes (aspnet/Mvc#6870, aspnet/Mvc#6784)
logic, timeouts and circuit breakers (aspnet/HttpClientFactory#42)
- [x] Build-time Razor #5408 - @scottaddie -Compile Razor pages and views as part of your build, improved startup performance (aspnet/Razor#1809)
- [x] Razor pages
   * - [x] Support for areas #5407 - @Rick-Anderson
   * - [x]  ~/Pages/Shared (aspnet/Mvc#6926, aspnet/Mvc#6604) also described by `@DamianEdwards` on the Standup :point_right: https://youtu.be/-EgE79M3tuc?t=36m20s See [this](https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/razor-pages-start?view=aspnetcore-2.1#the-pagesshared-folder)
   * - [x]  **CUT from V2.1** Bind all props on a page or controller :point_right: https://youtu.be/-EgE79M3tuc?t=41m10s
   * - [x] Decorate simple properties as `[Required]` #5834 - @Rick-Anderson (will be in preview2) :point_right: https://youtu.be/-EgE79M3tuc?t=45m25s

---------------- Issues not in aspnet/Announcements#288 -------

@guardrex

- [x] Update Kestrel host configuration guidance to include new HostFiltering middleware 
- [x] Update note regarding using ASP.NET Core Previews on Azure App Service #5568 
- [x] Sample Update Tracking Issue #5495 (The initial assignment is complete; work continues on the issue.)
- [x] 2.1 Update config for 2.x Azure KV config provider topic samples #4211 
- [x] Document improved support for integration testing #4652 
- [x] Hosting non-server apps with Generic Host  #5798 
- [x] Update doc for IHostedService #3352 
- [x] New kestrel configuration & related APIs #5056
- [x] Add HTTPS setup instructions for IIS Dev scenario #5899 
- [x] Add AllowedHosts to proxy config topic #5793
- [x] Update libuv references in Kestrel docs #5695
- [x] RP conventions features update #5520
- [x] Add KeyVault encryption to DataProtection #5439 
- [x] Change ANCM requestTimeout property ... #5264
- [x] MVC: Document ViewData-backed properties #6123
- [x] Document how Razor Pages handles HEAD requests #6124
- [x] Document updates to launchSettings.json #6127
- [x] Document Kestrel support for SNI #6129
- [x] Enhance DP cert protect/unprotect keys coverage #7358 
- [x] Razor Pages Tutorial - Seed Data as Migration #7342 
- [x] Update 'dev-certs' tools link in Kestrel server topic #6097 
- [ ] Update BuildWebHost to CreateWebHostBuilder (MD topics only) #6091 :bell: **IN REVIEW**
- [ ] Sample Update Tracking Issue #5495 🏃 **IN PROGRESS**
- [x] Replace M.A.All meta-package with Microsoft.AspNetCore.App #5314 
.....
- [x] Add Microsoft.AspNetCore.App to dotnet docs repo #5937 @rick-anderson
- [x]  Partial tag helper analyzer warnings #5938 @rick-anderson
- [x]  Document SetCompatibilityVersion #5939 @rick-anderson
- [x] Document `IPageFilter` and filter methods on `PageModel`  #5754-- contact @pranavkm Related remarks in the Standup :point_right: :point_right: https://youtu.be/-EgE79M3tuc?t=43m16s    * Code that runs before or after every handler (`OnPageHandlerExecuting`/`OnPageHandlerExecuted`) :point_right: https://youtu.be/-EgE79M3tuc?t=43m16s
- [x] Document Microsoft.AspNetCore.app #5913 @rick-anderson
- [x] **Partial Tag Helper** #5420 -- https://github.com/aspnet/Mvc/issues/5916
- [ ] Update the filters documentation to include info about the IAlwaysRunFilter and IAsyncAlwaysRunFilter filters #5419 - @Rick-Anderson
- [x] Update security/authentication/identity-enable-qrcodes for 2.1 changes #5487
- [ ] Doc new IOptionsBuilder and related overloads #5924

**Comment by pranavkm** [comment]
@Rick-Anderson we're missing docs for `IPageFilter` and filter methods on `PageModel`. Would this be an appropriate issue to track these docs?

**Comment by Rick-Anderson** [comment]
@pranavkm Thanks! Added to the list and put you down as the contact. Do we have any information to get us jump started?

**Comment by pranavkm** [comment]
I wrote some in the same vein as `IActionFilter`: https://github.com/aspnet/Home/issues/2981#issuecomment-374706661

**Comment by Rick-Anderson** [comment]
Closing as this doesn't have much useful info.
